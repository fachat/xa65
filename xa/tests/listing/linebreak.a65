
	*=$1000

	// default listbytes is 8
  
	lda #$0
	
	// first without labels

	.(
	// less than 8 bytes
	.byt 0,1,2,3,4,5	; 6
	// seven bytes
	.byt 0,1,2,3,4,5,6	; 7
	// eight bytes
	.byt 0,1,2,3,4,5,6,7	; 8
	// nine bytes
	.byt 0,1,2,3,4,5,6,7,8	; 9
	// ten bytes
	.byt 0,1,2,3,4,5,6,7,8,9	; 10
	// eleven bytes
	.byt 0,1,2,3,4,5,6,7,8,9,10	; 11
	// fifteen bytes
	.asc "012345678901234"	; 15, asc test as well
	// sixteen bytes
	.asc "0123456789012345"	; 16
	// seventeen bytes
	.asc "01234567890123456"; 17
	// thirtythree
	.asc "012345678901234567890123456789012"; 32

	// now with labels

	// less than 8 bytes
l6	.byt 0,1,2,3,4,5	; 6
	// seven bytes
l7	.byt 0,1,2,3,4,5,6	; 7
	// eight bytes
l8	.byt 0,1,2,3,4,5,6,7	; 8
	// nine bytes
l9	.byt 0,1,2,3,4,5,6,7,8	; 9
	// ten bytes
l10	.byt 0,1,2,3,4,5,6,7,8,9	; 10
	// eleven bytes
l11	.byt 0,1,2,3,4,5,6,7,8,9,10	; 11
	// fifteen bytes
l15	.asc "012345678901234"	; 15, asc test as well
	// sixteen bytes
l16	.asc "0123456789012345"	; 16
	// seventeen bytes
l17	.asc "01234567890123456"; 17
	// thirtythree
l32	.asc "012345678901234567890123456789012"; 32

	.)

	// now set number of bytes per listing line to 10

	.listbytes 10

	.(
	// less than 8 bytes
	.byt 0,1,2,3,4,5	; 6
	// seven bytes
	.byt 0,1,2,3,4,5,6	; 7
	// eight bytes
	.byt 0,1,2,3,4,5,6,7	; 8
	// nine bytes
	.byt 0,1,2,3,4,5,6,7,8	; 9
	// ten bytes
	.byt 0,1,2,3,4,5,6,7,8,9	; 10
	// eleven bytes
	.byt 0,1,2,3,4,5,6,7,8,9,10	; 11
	// fifteen bytes
	.asc "012345678901234"	; 15, asc test as well
	// sixteen bytes
	.asc "0123456789012345"	; 16
	// seventeen bytes
	.asc "01234567890123456"; 17
	// thirtythree
	.asc "012345678901234567890123456789012"; 32


	// now with labels

	// less than 8 bytes
l6	.byt 0,1,2,3,4,5	; 6
	// seven bytes
l7	.byt 0,1,2,3,4,5,6	; 7
	// eight bytes
l8	.byt 0,1,2,3,4,5,6,7	; 8
	// nine bytes
l9	.byt 01,2,3,4,5,6,7,8	; 9
	// ten bytes
l10	.byt 01,2,3,4,5,6,7,8,9	; 10
	// eleven bytes
l11	.byt 01,2,3,4,5,6,7,8,9,10	; 11
	// fifteen bytes
l15	.asc "012345678901234"	; 15, asc test as well
	// sixteen bytes
l16	.asc "0123456789012345"	; 16
	// seventeen bytes
l17	.asc "01234567890123456"; 17
	// thirtythree
l32	.asc "012345678901234567890123456789012"; 32

	.)

	// set number of listing bytes to unlimited
	.listbytes unlimited

	.(
	// less than 8 bytes
	.byt 0,1,2,3,4,5	; 6
	// seven bytes
	.byt 0,1,2,3,4,5,6	; 7
	// eight bytes
	.byt 0,1,2,3,4,5,6,7	; 8
	// nine bytes
	.byt 0,1,2,3,4,5,6,7,8	; 9
	// ten bytes
	.byt 0,1,2,3,4,5,6,7,8,9	; 10
	// eleven bytes
	.byt 0,1,2,3,4,5,6,7,8,9,10	; 11
	// fifteen bytes
	.asc "012345678901234"	; 15, asc test as well
	// sixteen bytes
	.asc "0123456789012345"	; 16
	// seventeen bytes
	.asc "01234567890123456"; 17
	// thirtythree
	.asc "012345678901234567890123456789012"; 32


	// now with labels

	// less than 8 bytes
l6	.byt 0,1,2,3,4,5	; 6
	// seven bytes
l7	.byt 0,1,2,3,4,5,6	; 7
	// eight bytes
l8	.byt 0,1,2,3,4,5,6,7	; 8
	// nine byte
l9	.byt 0,1,2,3,4,5,6,7,8	; 9
	// ten bytes
l10	.byt 0,1,2,3,4,5,6,7,8,9	; 10
	// eleven bytes
l11	.byt 0,1,2,3,4,5,6,7,8,9,10	; 11
	// fifteen bytes
l15	.asc "012345678901234"	; 15, test as well
	// sixteen bytes
l16	.asc "0123456789012345"	; 16
	// seventeen bytes
l17	.asc "01234567890123456"; 17
	// thirtythree
l32	.asc "012345678901234567890123456789012"; 32

	.)

		
