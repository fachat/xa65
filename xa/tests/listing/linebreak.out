
linebreak.a65


    2 A:1000                                     *= $1000

    6 A:1000  a9 00                              lda #$00

   10 A:1002                                     .( 

   12 A:1002  00 01 02 03 04 05                  .byt 0,1,2,3,4,5; 6

   14 A:1008  00 01 02 03 04 05 06               .byt 0,1,2,3,4,5,6; 7

   16 A:100f  00 01 02 03 04 05 06 07            .byt 0,1,2,3,4,5,6,7; 8

   18 A:1017  00 01 02 03 04 05 06 ...           .byt 0,1,2,3,4,5,6,7,8; 9

   20 A:1020  00 01 02 03 04 05 06 ...           .byt 0,1,2,3,4,5,6,7,8,9; 10

   22 A:102a  00 01 02 03 04 05 06 ...           .byt 0,1,2,3,4,5,6,7,8,9,10; 11

   24 A:1035  30 31 32 33 34 35 36 ...           .asc "012345678901234"  ; 15, asc test as well

   26 A:1044  30 31 32 33 34 35 36 ...           .asc "0123456789012345" ; 16

   28 A:1054  30 31 32 33 34 35 36 ...           .asc "01234567890123456"; 17

   30 A:1065  30 31 32 33 34 35 36 ...           .asc "012345678901234567890123456789012"; 32

   35 A:1086  00 01 02 03 04 05        l6        .byt 0,1,2,3,4,5; 6

   37 A:108c  00 01 02 03 04 05 06     l7        .byt 0,1,2,3,4,5,6; 7

   39 A:1093  00 01 02 03 04 05 06 07  l8        .byt 0,1,2,3,4,5,6,7; 8

   41 A:109b  00 01 02 03 04 05 06 ... l9        .byt 0,1,2,3,4,5,6,7,8; 9

   43 A:10a4  00 01 02 03 04 05 06 ... l10       .byt 0,1,2,3,4,5,6,7,8,9; 10

   45 A:10ae  00 01 02 03 04 05 06 ... l11       .byt 0,1,2,3,4,5,6,7,8,9,10; 11

   47 A:10b9  30 31 32 33 34 35 36 ... l15       .asc "012345678901234"; 15, asc test as well

   49 A:10c8  30 31 32 33 34 35 36 ... l16       .asc "0123456789012345"; 16

   51 A:10d8  30 31 32 33 34 35 36 ... l17       .asc "01234567890123456"; 17

   53 A:10e9  30 31 32 33 34 35 36 ... l32       .asc "012345678901234567890123456789012"; 32

   55 A:110a                                     .) 

   59 A:110a                                     .listbytes 10

   61 A:110a                                     .( 

   63 A:110a  00 01 02 03 04 05                  .byt 0,1,2,3,4,5; 6

   65 A:1110  00 01 02 03 04 05 06               .byt 0,1,2,3,4,5,6; 7

   67 A:1117  00 01 02 03 04 05 06 07            .byt 0,1,2,3,4,5,6,7; 8

   69 A:111f  00 01 02 03 04 05 06 07 08         .byt 0,1,2,3,4,5,6,7,8; 9

   71 A:1128  00 01 02 03 04 05 06 07 08 09      .byt 0,1,2,3,4,5,6,7,8,9; 10

   73 A:1132  00 01 02 03 04 05 06 07 08 ...     .byt 0,1,2,3,4,5,6,7,8,9,10; 11

   75 A:113d  30 31 32 33 34 35 36 37 38 ...     .asc "012345678901234"  ; 15, asc test as well

   77 A:114c  30 31 32 33 34 35 36 37 38 ...     .asc "0123456789012345" ; 16

   79 A:115c  30 31 32 33 34 35 36 37 38 ...     .asc "01234567890123456"; 17

   81 A:116d  30 31 32 33 34 35 36 37 38 ...     .asc "012345678901234567890123456789012"; 32

   87 A:118e  00 01 02 03 04 05        l6        .byt 0,1,2,3,4,5; 6

   89 A:1194  00 01 02 03 04 05 06     l7        .byt 0,1,2,3,4,5,6; 7

   91 A:119b  00 01 02 03 04 05 06 07  l8        .byt 0,1,2,3,4,5,6,7; 8

   93 A:11a3  01 02 03 04 05 06 07 08  l9        .byt 1,2,3,4,5,6,7,8; 9

   95 A:11ab  01 02 03 04 05 06 07 08 09 
   95 A:11ab                           l10       .byt 1,2,3,4,5,6,7,8,9; 10

   97 A:11b4  01 02 03 04 05 06 07 08 09 0a 
   97 A:11b4                           l11       .byt 1,2,3,4,5,6,7,8,9,10; 11

   99 A:11be  30 31 32 33 34 35 36 37 38 ...
   99 A:11be                           l15       .asc "012345678901234"; 15, asc test as well

  101 A:11cd  30 31 32 33 34 35 36 37 38 ...
  101 A:11cd                           l16       .asc "0123456789012345"; 16

  103 A:11dd  30 31 32 33 34 35 36 37 38 ...
  103 A:11dd                           l17       .asc "01234567890123456"; 17

  105 A:11ee  30 31 32 33 34 35 36 37 38 ...
  105 A:11ee                           l32       .asc "012345678901234567890123456789012"; 32

  107 A:120f                                     .) 

  110 A:120f                                     .listbytes 0

  112 A:120f                                     .( 

  114 A:120f  00 01 02 03 04 05                  .byt 0,1,2,3,4,5; 6

  116 A:1215  00 01 02 03 04 05 06               .byt 0,1,2,3,4,5,6; 7

  118 A:121c  00 01 02 03 04 05 06 07            .byt 0,1,2,3,4,5,6,7; 8

  120 A:1224  00 01 02 03 04 05 06 07 08         .byt 0,1,2,3,4,5,6,7,8; 9

  122 A:122d  00 01 02 03 04 05 06 07 08 09      .byt 0,1,2,3,4,5,6,7,8,9; 10

  124 A:1237  00 01 02 03 04 05 06 07 08 09 0a   .byt 0,1,2,3,4,5,6,7,8,9,10; 11

  126 A:1242  30 31 32 33 34 35 36 37 38 39 30 31 32 33 34 
  126 A:1242                                     .asc "012345678901234"  ; 15, asc test as well

  128 A:1251  30 31 32 33 34 35 36 37 38 39 30 31 32 33 34 35 
  128 A:1261                                     .asc "0123456789012345" ; 16

  130 A:1261  30 31 32 33 34 35 36 37 38 39 30 31 32 33 34 35 
  130 A:1271  36                                 .asc "01234567890123456"; 17

  132 A:1272  30 31 32 33 34 35 36 37 38 39 30 31 32 33 34 35 
  132 A:1282  36 37 38 39 30 31 32 33 34 35 36 37 38 39 30 31 
  132 A:1292  32                                 .asc "012345678901234567890123456789012"; 32

  138 A:1293  00 01 02 03 04 05        l6        .byt 0,1,2,3,4,5; 6

  140 A:1299  00 01 02 03 04 05 06     l7        .byt 0,1,2,3,4,5,6; 7

  142 A:12a0  00 01 02 03 04 05 06 07  l8        .byt 0,1,2,3,4,5,6,7; 8

  144 A:12a8  00 01 02 03 04 05 06 07 08 
  144 A:12a8                           l9        .byt 0,1,2,3,4,5,6,7,8; 9

  146 A:12b1  00 01 02 03 04 05 06 07 08 09 
  146 A:12b1                           l10       .byt 0,1,2,3,4,5,6,7,8,9; 10

  148 A:12bb  00 01 02 03 04 05 06 07 08 09 0a 
  148 A:12bb                           l11       .byt 0,1,2,3,4,5,6,7,8,9,10; 11

  150 A:12c6  30 31 32 33 34 35 36 37 38 39 30 31 32 33 34 
  150 A:12c6                           l15       .asc "012345678901234"; 15, test as well

  152 A:12d5  30 31 32 33 34 35 36 37 38 39 30 31 32 33 34 35 
  152 A:12e5                           l16       .asc "0123456789012345"; 16

  154 A:12e5  30 31 32 33 34 35 36 37 38 39 30 31 32 33 34 35 
  154 A:12f5  36                       l17       .asc "01234567890123456"; 17

  156 A:12f6  30 31 32 33 34 35 36 37 38 39 30 31 32 33 34 35 
  156 A:1306  36 37 38 39 30 31 32 33 34 35 36 37 38 39 30 31 
  156 A:1316  32                       l32       .asc "012345678901234567890123456789012"; 32

  158 A:1317                                     .) 
